### DISCUSSION TITLE
Support for Nuxt and other frameworks

### DISCUSSION THREAD

**Discussion by OmarMcAdam** [triggerdotdev/trigger.dev#411] [Ideas]
Trigger.dev has potential to serve as a unified task runner across all JS frameworks
It would be great to know what dependancies trigger has to react/next and make a list of what would be needed for it to become framework agnostic.

**Comment by ericallam** [comment]
Thanks for the idea! This is a good opportunity to initially document everything it would take to "properly" support another JS framework, in no particular order:

- [ ] Create an "adapter" package that exposes a `TriggerClient` on the server at `/api/trigger` or similar path. [Next.js adapter](https://github.com/triggerdotdev/trigger.dev/tree/main/packages/nextjs)
- [ ] If not a React framework, create an equivalent of the `@trigger.dev/react` package ([reference](https://github.com/triggerdotdev/trigger.dev/tree/main/packages/react)) that makes it easy to get public info about jobs/runs/tasks from the frontend
- [ ] Add to the `@trigger.dev/cli init` command to automatically detect framework and bootstrap it with Trigger.dev code
    - [ ] If this is the first framework after Next.js, we'll need to modify the `init` command code to support multiple framework detection
- [ ] Add a "Quick Start" guide for the framework to the docs [Next.js Quick Start](https://trigger.dev/docs/documentation/quickstarts/nextjs)
- [ ] Add a Platform guide for the framework to the docs [Next.js Platform Guide](https://trigger.dev/docs/documentation/guides/platforms/nextjs)
- [ ] Add the framework to the dashboard onboarding tutorial
    - [ ] If this is the first framework after Next.js, we'll need to update the onboarding tutorial to support multiple frameworks
- [ ] Add an example project to the examples repo [ref](https://github.com/triggerdotdev/examples)

**Comment by harshmangalam** [comment]
@ericallam If you need any help to add Qwik/Qwikcity support i am ready.

**Comment by matt-aitken** [comment]
I'm closing this as v3 doesn't require framework specific code to run tasks and is now available in early access: https://trigger.dev/blog/v3-developer-preview-launch/
